### DISCUSSION TITLE
saving rating and playlist

### DISCUSSION THREAD

**Discussion by wotori** [tramhao/termusic#88] [Ideas]
How about adding rating feature to allow mark favourite songs and save playlist?
I also wonder if it is possible to sync with mobile device, transfer marked songs somewhere... for these task I guess at first its required to generate a list of file paths and then use this list to copy into specific directory (mobile device or remote server). This location could be added to option screen that appears after ctrl+c shortcut.

Last days I imagine smth like this pipeline: store entire library on personal local or remote server and to be able to play songs from It while at home and always have a local copy of all starred songs. Does this idea reasonable and could it be implemented/integrated into project?

**Comment by tramhao** [comment]
Good ideas. Let me put it to my todos.
About sync, I use verysync, and it has linux/windows/android/openwrt/nas client. I think it's similar to resilio sync.
But your ideas about syncing is also very helpful. I need to think about the implementation.

**Reply by wotori** [reply]
Mention [termscp](https://github.com/veeso/termscp). Probably It could be used as dependency for syncing playlists with remote server and vice versa.

**Comment by tramhao** [comment]
Found a similar project https://github.com/bytesnake/hex. Sound quite promising!

**Comment by hasezoey** [comment]
This discussion is about 2 things: rating(favorites) and syncing songs. Lets keep it to just ratings(favorites) for now, and if wanted, please open another discussion.

As for ratings, how would you think to implement it? For me 2 things would come to mind (plus a hybrid version):
- ratings on a scale (lets say 0(unrated) to 10, please suggest a scale)
- simple boolean favorites (yes/no)
- hybrid: allow ratings on a scale, but add a options for simple favorites

The "simple boolean" version is likely the most commonly implemented, and the thing i would go for (as i personally have no need for anything in-between. I would just make multiple playlists if i wanted some specific collection)
